### PR TITLE
Enable --strictPropertyInitialization

### PR DESCRIPTION
**Pull request by andy-ms**
Enables `--strictPropertyInitialization` in `tsconfig-base.json` and updates declarations that were now errors.

### CODE REVIEW

File: src/server/project.ts [hunk 1]
```diff
@@ -896,7 +896,7 @@ namespace ts.server {
 
         /* @internal */
         getCurrentProgram() {
-            return this.program;
+            return Debug.assertDefined(this.program);
```
Comment (sheetalkamat) [inline src/server/project.ts]: I don't think this is always defined? This can be undefined

File: src/server/editorServices.ts [hunk 2]
```diff
@@ -470,7 +470,7 @@ namespace ts.server {
 
         private pendingProjectUpdates = createMap<Project>();
         /* @internal */
-        pendingEnsureProjectForOpenFiles: boolean;
+        pendingEnsureProjectForOpenFiles: boolean | undefined;
```
Comment (sheetalkamat) [inline src/server/editorServices.ts]: = false instead ?

File: src/server/project.ts [hunk 3]
```diff
@@ -1312,20 +1312,20 @@ namespace ts.server {
...
         /* @internal */
-        pendingReload: ConfigFileProgramReloadLevel;
+        pendingReload: ConfigFileProgramReloadLevel | undefined;
 
         /*@internal*/
         configFileSpecs: ConfigFileSpecs | undefined;
 
         /*@internal*/
-        canConfigFileJsonReportNoInputFiles: boolean;
+        canConfigFileJsonReportNoInputFiles: boolean | undefined;
```
Comment (sheetalkamat) [inline src/server/project.ts]: = false?

File: src/server/scriptInfo.ts [hunk 4]
```diff
@@ -248,7 +248,7 @@ namespace ts.server {
         private realpath: Path | undefined;
 
         /*@internal*/
-        cacheSourceFile: DocumentRegistrySourceFileCache;
+        cacheSourceFile!: DocumentRegistrySourceFileCache; // TODO: GH#18217
```
Comment (sheetalkamat) [inline src/server/scriptInfo.ts]: Should be instead `DocumentRegistrySourceFileCache | undefined`

File: src/services/shims.ts [hunk 5]
```diff
@@ -485,18 +485,24 @@ namespace ts {
...
         public realpath: (path: string) => string;
         public useCaseSensitiveFileNames: boolean;
 
         constructor(private shimHost: CoreServicesShimHost) {
-        this.useCaseSensitiveFileNames = this.shimHost.useCaseSensitiveFileNames ? this.shimHost.useCaseSensitiveFileNames() : false;
+            this.useCaseSensitiveFileNames = this.shimHost.useCaseSensitiveFileNames ? this.shimHost.useCaseSensitiveFileNames() : false;
             if ("directoryExists" in this.shimHost) {
                 this.directoryExists = directoryName => this.shimHost.directoryExists(directoryName);
             }
+            else {
```
Comment (sheetalkamat) [inline src/services/shims.ts]: why?

Comment (andy-ms) [reply src/services/shims.ts]: `TypingResolutionHost` requires `directoryExists` to be non-optional. It's likely that `this.shimHost` always contains `directoryExists`, so it always gets defined.

**Comment by andy-ms** [comment]
@sheetalkamat Please re-review

**Pull request merged**